### PR TITLE
Vscode Vertx Starter blog post

### PR DESCRIPTION
**Pull request by danielpetisme**
As discussed a blog post to announce Visual Studio Code Extension.

The extension is already published on the marketplace so the date of publication doesn't matter (I saw the 3.7.1 realease for 27/05, I think the publications shouldn't be on the same period.

The extension
https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter

### CODE REVIEW

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+template: post.html
+author: danielpetisme
+draft: true
+---
+
+> If Visual Studio Code is on the hype, Vert.x is on the edge!
+
+We are really excited to announce the [Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) extension.
+
+![Screenshot](https://github.com/danielpetisme/vscode-vertx-starter/raw/master/images/vscode-vertx-starter.gif)
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: Do you have the capture in video format? A link to a Youtube/Vimeo/... video would be preferable to loading a ~16M gif.

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+title: VS Code Vertx Starter Extension
+date: 2019-05-27
+template: post.html
+author: danielpetisme
+draft: true
+---
+
+> If Visual Studio Code is on the hype, Vert.x is on the edge!
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: I'm not convinced this is needed.

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+We are really excited to announce the [Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) extension.
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: Can you reword to explain who you are and this is a community contrib?

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+![Screenshot](https://github.com/danielpetisme/vscode-vertx-starter/raw/master/images/vscode-vertx-starter.gif)
+
+Visual Studio Code is the living representation of Microsoft cultural shift. It's a well appreciate Text Editor/IDE and thanks to the extension developement's simplicity more and more actors are proposing new features.
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: Can you rephrase the first sentence. It's enough to say it's popular, I think.

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+[Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) permits your to:
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]:
```suggestion
[Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) allows your to:
```

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+- Create a Vert.x project
+- Customize the creation (Vert.x version, language, build tool, groupId, artifactId, package name, JDK version)
+- Search for dependencies
+
+The extension can be configured to match your context (E.g. defining your default company groupId).
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]:
```suggestion
The extension can be configured to match your context (e.g. defining your default company `groupId`).
```

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+We are really excited to announce the [Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) extension.
+
+![Screenshot](https://github.com/danielpetisme/vscode-vertx-starter/raw/master/images/vscode-vertx-starter.gif)
+
+Visual Studio Code is the living representation of Microsoft cultural shift. It's a well appreciate Text Editor/IDE and thanks to the extension developement's simplicity more and more actors are proposing new features.
+
+[Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) permits your to:
+- Create a Vert.x project
+- Customize the creation (Vert.x version, language, build tool, groupId, artifactId, package name, JDK version)
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]:
```suggestion
- Customize the creation (Vert.x version, language, build tool, `groupId`, `artifactId`, package name, JDK version)
```

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+- Search for dependencies
+
+The extension can be configured to match your context (E.g. defining your default company groupId).
+
+Behind the scene, the extension simply relies on the Vert.x application generator [start.vertx.io](https://start.vertx.io) API and demonstrate how easy it is to develop your own IDE plugin or custom tooling to generate Vert.x projects. Further information on [Vert.x Starter GitHub project](https://github.com/vert-x3/vertx-starter).
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]:
```suggestion
Behind the scene, the extension relies on the Vert.x application generator API and demonstrate how easy it is to develop your own IDE plugin or custom tooling to generate Vert.x projects. 

For further details about the [start.vertx.io](https://start.vertx.io) API, please refer to the [Vert.x Starter project on GitHub](https://github.com/vert-x3/vertx-starter).
```

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+We would love to have your feeback on our user or dev [channels](https://vertx.io/community).
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: Can you point to the extension repo instead?

File: src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+template: post.html
+author: danielpetisme
+draft: true
+---
+
+Hi Vert.x community! Today, we are really excited to announce the [Visual Studio Code Vert.x Starter](https://marketplace.visualstudio.com/items?itemName=danielpetisme.vscode-vertx-starter) extension.
+
+The extension is a community contribution made by [Daniel Petisme (@danielpetisme)](https://twitter.com/danielpetisme).
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/R3ZbQ_5Jf4M" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (tsegismont) [inline src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: Have you checked the result locally?

Comment (tsegismont) [reply src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: I mean does the video shows up correctly on the web page?

Comment (danielpetisme) [reply src/site/blog/posts/releases/2019-05-27-vscode-vertx-starter.md]: I checked with:
https://jbt.github.io/markdown-editor/
Looks good, do you know another way to test it ?

**Comment by danielpetisme** [comment]
Do you want me to change the publication date and draft status?
---
title: VS Code Vert.x Starter Extension
date: 2019-05-27
template: post.html
author: danielpetisme
draft: true
---

**Comment by tsegismont** [comment]
@danielpetisme I meant building the website locally

https://github.com/vert-x3/vertx-web-site/blob/master/README.md#building

**Comment by vietj** [comment]
great @danielpetisme let's publish that in the middle/end of next week

**Comment by danielpetisme** [comment]
@tsegismont 😢 
```
~/workspace/vertx-web-site (vscode-vertx-starter) mvn clean site

[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< io.vertx:vertx-web-site >-----------------------
[INFO] Building Vert.x Website 3.7.0
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from JitPack: https://jitpack.io/com/github/mauricio/postgresql-async_2/12/db-async-common_2.12/0.2.21/db-async-common_2.12-0.2.21.pom
Downloading from JitPack: https://jitpack.io/com/github/mauricio/postgresql-async_2/12/mysql-async_2.12/0.2.21/mysql-async_2.12-0.2.21.pom
Downloading from JitPack: https://jitpack.io/com/github/mauricio/postgresql-async_2/12/postgresql-async_2.12/0.2.21/postgresql-async_2.12-0.2.21.pom
[WARNING] The POM for com.github.jknack:handlebars:jar:4.1.0 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.207 s
[INFO] Finished at: 2019-05-24T21:10:45+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project vertx-web-site: Could not resolve dependencies for project io.vertx:vertx-web-site:jar:3.7.0: Failed to collect dependencies at io.vertx:vertx-stack-docs:zip:docs:3.7.0 -> io.vertx:vertx-mysql-postgresql-client:zip:docs:3.7.0 -> com.github.mauricio:postgresql-async_2.12:jar:0.2.21 -> com.github.mauricio.postgresql-async_2.12:db-async-common_2.12:jar:0.2.21: Failed to read artifact descriptor for com.github.mauricio.postgresql-async_2.12:db-async-common_2.12:jar:0.2.21: Could not transfer artifact com.github.mauricio.postgresql-async_2.12:db-async-common_2.12:pom:0.2.21 from/to JitPack (https://jitpack.io): Not authorized -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

**Comment by tsegismont** [comment]
@danielpetisme not sure why it fails on your machine. Works for me. Anyway, embedding video was fine from a technical standpoint, but I had a second thought: embedding YouTube content will bring cookies ..etc without explicit consent from the readers. So I replaced it with a clickable image to YouTube site.

Thanks for this great contrib, I will publish the blog on Wednesday.

**Comment by danielpetisme** [comment]
Nice catch for the cookies!
Thanks
